src/SystemManager.cc

```diff
@@ -403,6 +404,19 @@ void SystemManager::ProcessPendingEntitySystems()
        continue;
     }
 
+    // set to world entity if entity id is not specified in the request.
+    if (entity == kNullEntity || entity == 0u)
+    {
+      entity = worldEntity(*this->entityCompMgr);
+    }
+    // otherwise check if entity exists before attempting to load the plugin.
+    else if (!this->entityCompMgr->HasEntity(entity))
+    {
+      gzwarn << "Unable to add plugins to Entity: '" << entity
```

Comment:
Might be worth setting to `gzerr` as by default end users don't see any indication of failure unless the enable logging at higher verbosities. In addition the add service should probable return false if we fail to add a object, but by default it always returns true.

Reply:
updated this msg and the one above to `gzerr`. [510a5dd](https://github.com/gazebosim/gz-sim/pull/2516/commits/510a5ddbe85bd4d0d4b18b93dac54a20966e2022)

> In addition the add service should probable return false if we fail to add a object, but by default it always returns true.

I noticed we do that in several places in gz-sim so that the requests can be handled asynchronously. The response data `true` indicates that the request was handled but not necessarily successful. I added a todo note on this. Not sure if there's a way to set response data in an asynchronous way, maybe @caguero knows if it's currently possible?